course3/homework-unique_ptr/test.cc

```diff
@@ -0,0 +1,99 @@
+#include "unique_ptr.h"
+#include "unique_ptr.cc"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+
+
+template<template<class...> class UniquePtrType>
+void unique_ptr_test()
+{
+	// default contructor
+	UniquePtrType<int> ptr;
+	assert(!ptr);
+	assert(ptr.get() == nullptr);
+	
+	// initialization constructor
+	UniquePtrType<int> ptr2(nullptr);
+	assert(!ptr2);
+	
+	int* i = new int(42);
+	UniquePtrType<int> ptr3(i);
+	assert(ptr3);
+	assert(*ptr3.get() == 42);
```

Comment:
get shouldn't be required. a smart pointer should mimic the syntax of raw pointer. so ```assert(*ptr3 == 42);```